### PR TITLE
feat(studio): add model selector to Safe Synthesizer creation [ASTD-19]

### PR DESCRIPTION
**Pull request by marcusds**
## Summary

Adds a **model** dropdown under **Training Data Configuration** in the Safe Synthesizer creation flow (ASTD-19). Bound to `spec.config.training.pretrained_model`.

Options (in ticket order):
1. `HuggingFaceTB/SmolLM3-3B` (default)
2. `mistralai/Mistral-7B-Instruct-v0.3`
3. `TinyLlama/TinyLlama-1.1B-Chat-v1.0`

Info tooltip: "Model that gets fine-tuned to generate synthetic data."

## Changes
- `constants.ts` — `PRETRAINED_MODEL_OPTIONS` + `DEFAULT_PRETRAINED_MODEL`.
- `schema.ts` — added `pretrained_model` to the training config + default. Flows into the submitted job payload automatically.
- `AdvancedParameters.tsx` — `ControlledSelect` at the top of the Training Data Configuration section.
- `AdvancedParameters.test.tsx` — tests for render, default (SmolLM3), option order, and selection updating the config.

## Acceptance criteria
- [x] Selector shown under Training Data Configuration
- [x] Three options, in the specified order
- [x] Defaults to SmolLM3-3B when untouched
- [x] Saved config reflects the selection

## Note
Backend `pretrained_model` is a free-form string; only **SmolLM3-3B** is currently pre-provisioned as a model fileset (`plugins/nemo-safe-synthesizer/scripts/setup_model_filesets.py`). Mistral-7B and TinyLlama resolve via HF at job time and may require provisioning before they run reliably.

## Testing
`pnpm --filter nemo-studio-ui test src/routes/SafeSynthesizerNewRoute` → 71 passed.

## Summary by CodeRabbit

* **New Features**
  * Added a pretrained model selector to the Safe Synthesizer “Training Data Configuration”.
  * The selector offers three supported options and applies a default when none is chosen.
  * The selected pretrained model is included in the saved synthesizer configuration.

* **Bug Fixes**
  * Ensured the pretrained model field is part of schema validation and form default behavior.

* **Tests**
  * Expanded coverage for the default selection, available options order, and updating the watched value after user changes.

### CODE REVIEW

File: web/packages/studio/src/routes/SafeSynthesizerNewRoute/schema.ts [hunk 1]
```diff
@@ -100,6 +103,7 @@ export const safeSynthesizerJobRequestSchema = z.object({
               }
             ),
           rope_scaling_factor: z.union([z.literal('auto'), z.number().int().min(1).max(6)]),
+          pretrained_model: z.string().optional(),
```
[the remark this answers is not in the document]
Comment (marcusds) [reply web/packages/studio/src/routes/SafeSynthesizerNewRoute/schema.ts]: Claude says: "Applied — `pretrained_model` now validates against `PRETRAINED_MODEL_OPTIONS` via `z.enum` instead of accepting any string. Fixed in 945529a."

File: web/packages/studio/src/routes/SafeSynthesizerNewRoute/components/AdvancedParameters.tsx [hunk 2]
```diff
@@ -158,6 +162,34 @@ export const AdvancedParameters = () => {
...
+            name: 'spec.config.training.pretrained_model',
+            control,
+          }}
+          items={PRETRAINED_MODEL_OPTIONS.map((model) => ({
+            value: model,
+            children: model,
+          }))}
+          formFieldProps={{
+            slotLabel: (
+              <>
```
Comment (steramae-nvidia) [inline web/packages/studio/src/routes/SafeSynthesizerNewRoute/components/AdvancedParameters.tsx:176]: Should you wrap this in Text styled as label instead of a Fragment?

**Review by steramae-nvidia — APPROVED** [review]

File: web/packages/studio/src/routes/SafeSynthesizerNewRoute/components/AdvancedParameters.tsx [hunk 3]
```diff
@@ -158,6 +162,34 @@ export const AdvancedParameters = () => {
...
+                model
```
Comment (nakolean) [inline web/packages/studio/src/routes/SafeSynthesizerNewRoute/components/AdvancedParameters.tsx]: **Question**: Capitalize the label?
```suggestion
                Model
```

**Review by nakolean — APPROVED** [review]
I have mild concerns about hard coding specific models since they will likely need frequent updates, although I'm not sure how else we'd define it without the same issue of needing to manually update it frequently 🤷

File: web/packages/studio/src/routes/SafeSynthesizerNewRoute/components/AdvancedParameters.tsx (hunk 3 above)
Comment (marcusds) [reply web/packages/studio/src/routes/SafeSynthesizerNewRoute/components/AdvancedParameters.tsx]: Yeah its not the best, but this is the way it works right now. Hopefully in the future safe-synth can provide this.

File: web/packages/studio/src/routes/SafeSynthesizerNewRoute/components/AdvancedParameters.tsx (hunk 2 above)
Comment (marcusds) [reply web/packages/studio/src/routes/SafeSynthesizerNewRoute/components/AdvancedParameters.tsx:176]: already gets wrapped in KUI's <Label> by FormField
